src/System.Windows.Forms.Design/tests/UnitTests/ControlDesignerTests.cs

```
@@ -195,4 +198,78 @@ public void ControlDesigner_WndProc_InvokePaint_Success()
         };
         designer.TestAccessor().Dynamic.WndProc(ref m);
     }
+
+    [Fact]
+    public void ControlDesigner_AssociatedComponents_NullSite_Test()
+    {
+        using ControlDesigner controlDesigner = new();
+        using Control control = new();
+
+        using Control subControl = new();
+        controlDesigner.Initialize(control);
+
+        Assert.Empty(controlDesigner.AssociatedComponents);
+
+        control.Controls.Add(subControl);
+
+        Assert.Empty(controlDesigner.AssociatedComponents);
+    }
+
+    internal static Mock<ISite> CreateMockSiteWithDesignerHost(object designerHost)
+    {
+        Mock<ISite> mockSite = new(MockBehavior.Loose);
+        mockSite
+            .Setup(s => s.GetService(typeof(IDesignerHost)))
+            .Returns(designerHost);
+        mockSite
+            .Setup(s => s.GetService(typeof(IComponentChangeService)))
+            .Returns(null);
+        mockSite
+            .Setup(s => s.GetService(typeof(IInheritanceService)))
+            .Returns(null);
+        mockSite
+            .Setup(s => s.GetService(typeof(IDictionaryService)))
+            .Returns(null);
+        mockSite
+            .Setup(s => s.GetService(typeof(IExtenderListService)))
+            .Returns(null);
+        mockSite
+            .Setup(s => s.GetService(typeof(ITypeDescriptorFilterService)))
+            .Returns(null);
+        mockSite
+            .Setup(s => s.GetService(typeof(AmbientProperties)))
+            .Returns(null);
+        mockSite
+            .Setup(s => s.GetService(typeof(Control)))
+            .Returns(null);
+        mockSite
+            .SetupGet(s => s.Container)
+            .Returns((IContainer)null);
+
+        return mockSite;
+    }
+
+    [WinFormsFact]
+    public void ControlDesigner_AssociatedComponentsTest()
+    {
+        using Control control = new();
+        using ControlDesigner controlDesigner = new();
+
+        Mock<IDesignerHost> mockDesignerHost = new(MockBehavior.Loose);
```

Could you please use `Strict` behavior? We want to throw when we receive unexpected calls.